paddle/platform/CMakeLists.txt

```diff
@@ -1,5 +1,5 @@
 if(WITH_GPU)
-  cc_library(enforce SRCS enforce.cc DEPS nccl)
+  cc_library(enforce SRCS enforce.cc DEPS)
```

Comment:
I vaguely remember that NVIDIA guys have upgraded these things with CUDA9, cudnn7, NCCL2. 
You teach me that we need to call `group_start` before call `ALLReduce` NCCL calls. Is it right?

Reply:
Thank you for reminding me. I've added `group_start` and `group_end` in this PR.